### DISCUSSION TITLE
How can I get my site’s CMS global settings on every page, from the server-side, while caching for a period of time?

### DISCUSSION THREAD

**Discussion by mattjis** [vercel/next.js#30539] [Help]
I'm making a site for a client which wants to be able to edit basic things on their own.

Within their CMS I have controls for things such as theme, site title, notification bar text, header links and footer links.

But the catch is, I need to get these settings globally (on every single page) and on the server (client-side cannot access this API) all while keeping my API requests to a minimum (e.g caching for 10 mins at a time)

Do the new Next.js 12 updates help me achieve this in any way? Otherwise, how would I do this?

**Comment by zyzski** [comment]
you can fetch the data at the top level (app) then store it in an object that can be imported by other pages and used within getServerSideProps. if it's only needed on the client, it can be passed down via props or wrapped in context

**Reply by mattjis** [reply]
Yeah I feel like this is the only way, even though I'm really trying to prevent my pages being server side rendered — as it's not as quick.
